packages/server/src/sync/schema.ts

```diff
@@ -2,21 +2,39 @@ import { Channel } from '@botpress/messaging-channels'
 import Joi from 'joi'
 import { ReqSchema } from '../base/schema'
 
-const SyncWebhookSchema = Joi.object({
-  url: Joi.string().uri().required()
-})
-
 export const makeSyncRequestSchema = (channels: Channel[]) => {
-  const channelsSchema: { [name: string]: Joi.ObjectSchema<any> } = {}
+  const channelsSchema: { [name: string]: Joi.ObjectSchema } = {}
+  const channelsByName: { [name: string]: Channel[] } = {}
 
   for (const channel of channels) {
-    channelsSchema[channel.meta.name] = channel.meta.schema.optional()
+    if (!channelsByName[channel.meta.name]) {
+      channelsByName[channel.meta.name] = []
+    }
+    channelsByName[channel.meta.name].push(channel)
+  }
+
+  for (const [name, channels] of Object.entries(channelsByName)) {
+    channelsSchema[name] = Joi.object({
+      version: Joi.string()
+        .valid(...channels.map((x) => x.meta.version))
+        .optional()
+    })
+      .options({
+        allowUnknown: true
+      })
+      .optional()
   }
 
   return ReqSchema({
     body: {
-      channels: Joi.object(channelsSchema).allow(null),
-      webhooks: Joi.array().items(SyncWebhookSchema).allow(null)
+      channels: Joi.object(channelsSchema).optional(),
```

Comment:
The first round of validation only checks if we are providing valid webhooks, as well as only existing channel names with existing versions